stactask/task.py

```diff
@@ -108,15 +107,29 @@ def __init__(
 
     @property
     def process_definition(self) -> dict[str, Any]:
-        process = self._payload.get("process", {})
+        process = self._payload.get("process", [])
         if isinstance(process, dict):
+            warnings.warn(
+                "`process` is dictionary, use a list instead",
+                DeprecationWarning,
+                stacklevel=2,
+            )
             return process
-        else:
-            raise ValueError(f"process is not a dict: {type(process)}")
+
+        if not isinstance(process, list):
+            raise TypeError("`process` must be a list of dictionaries")
+
+        if not process:
+            return {}
+
+        if not isinstance(process[0], dict):
+            raise TypeError("`process` list elements must be dictionaries")
```

Comment:
This again is subtly inaccurate language. Only the first element in `process` must be a dict. As we saw above, elements can be arrays to represent branching, with the exception of the first element.

So maybe:

> unable to parse `process`: the first element of the list must be a dictionary

Reply:
Done.